### PR TITLE
Set disable registration as default true

### PR DESCRIPTION
**Pull request by sipsorcery**
For safety user registration is defaulted to true as per #455.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
If I do this, how does the first user can create the admin account?

**Comment by Kukks** [comment]
@sipsorcery  you need to enable the registration when there are no users registered at the very least then :)

**Comment by sipsorcery** [comment]
Adjusted so that registrations will always be allowed if there are no user records.

File: BTCPayServer/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -74,7 +74,7 @@
                         </li>}
                     else
                     {
-                        @if (themeManager.ShowRegister)
+                        @if (themeManager.ShowRegister || UserManager.Users.Count() == 0)
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Shared/_Layout.cshtml]: This mean we do one additional database call for every page if `ShowRegister` is false. Massive perf hit for a condition never satisfied.

**Comment by NicolasDorier** [comment]
@sipsorcery I would prefer you keep True by default, and when the admin is registered, you change the setting to lock subscriptions.

**Comment by sipsorcery** [comment]
@NicolasDorier PR is fixed to work as you recommended. 

For the tests are the instructions [here](https://github.com/btcpayserver/btcpayserver/blob/master/BTCPayServer.Tests/README.md) missing a step or two? `docker-compose up dev` reports that it can't find a valid configuration file. Has it been moved to the .circleci directory or is there some intermediate step needed to generate a docker-compose file?

**Comment by NicolasDorier** [comment]
@sipsorcery `docker-compose up dev` if you want to develop and run tests from visual studio.
`docker-compose run tests` run the tests from inside a container.

**Comment by NicolasDorier** [comment]
The tests still failing. You probably want to disable this feature for the tests.

**Comment by NicolasDorier** [comment]
@sipsorcery anytime to fix the tests ? I think this should definitively be merged once fixed.

**Comment by sipsorcery** [comment]
@NicolasDorier I was planning on fixing the tests once I’m back from Xmas travels (no computer at the moment).

**Comment by sipsorcery** [comment]
@NicolasDorier added `debug` environment check so this policy setting won't be applied when running unit tests. Seemed a bit better than an additional command line option.

**Comment by NicolasDorier** [comment]
@sipsorcery can you instead just add an option with the command line? like `--disable-registration`/`DisableRegistration` default to `true` and just set it in the `PayTester` for unit tests to `false`?

**Comment by NicolasDorier** [comment]
Awesome, I really wanted that.

**Pull request merged**